test/e2e_node/gpu_device_plugin.go

```diff
@@ -146,19 +150,28 @@ func restartKubelet(f *framework.Framework) {
 	kube := matches[0]
 	framework.Logf("Get running kubelet with systemctl: %v, %v", string(stdout), kube)
 	stdout, err = exec.Command("sudo", "systemctl", "restart", kube).CombinedOutput()
-	if err == nil {
-		return
+	if err != nil {
+		framework.Failf("Failed to restart kubelet with systemctl: %v, %v", err, stdout)
 	}
-	framework.Failf("Failed to restart kubelet with systemctl: %v, %v", err, stdout)
+	Eventually(func() bool {
+		afterSocks, err := filepath.Glob("/var/lib/kubelet/device-plugins/nvidiaGPU*.sock")
```

Comment:
Should we just check for the number of GPUs ? The socket is more of an implementation detail in the eyes of an end user

Reply:
Just checking no. of GPUs is not enough because the capacity update may fall behind.